lib/streamlit/delta_generator.py

```diff
@@ -362,7 +362,7 @@ def _parent_block_types(self) -> Iterable[str]:
         current_dg: Optional[DeltaGenerator] = self
         while current_dg is not None:
             if current_dg._block_type is not None:
-                yield current_dg._block_type
+                yield f"{current_dg._block_type}:{current_dg._root_container}"
```

Comment:
You are introducing a new undocumented contract that the `_parent_block_types` now returns a list of strings where each string consist of two parts, joined with `:` character. This can be pretty hard to read and also if someone will change the separator but forgets to update it 200 lines below we are in trouble. 

Instead I'd propose to change the method signature from:

```
_parent_block_types(self) -> Iterable[str]:
```

to:

```
_parent_block_types(self) -> Iterable[tuple[BlockType, RootContainer]]:
```

and then in `yield`, do:

```
yield current_dg._block_type, current_dg._root_container
```

Reply:
Thank you @sfc-gh-mnowotka for this review 🙇 You have a good point here 
In the commit [return iterable str (blocktype), int (rootcontainer) instead and move logic to functions](https://github.com/streamlit/streamlit/pull/5941/commits/51301260aa409f3931e480a3d792fe5d3c045957) I've introduced 3 functions and used them instead.

There is a slight difference to your proposal since `BlockType` is actually a `str` and `RootContainer` an `int`. (variable `RootContainer` is used as a protobuf proto value) I've created the following signature:
`Iterable[Tuple[str, Optional[int]]]` but the idea and `yield` is just as you suggested. Thank you it's a great advice 🙇